### PR TITLE
TravisCI setup

### PR DESCRIPTION
**Pull request by tagonzalez**
## Summary
 [Configurar TravisCI](https://trello.com/c/WNWqXW32/12-configurar-travisci)

Set up TravisCI for Continuous integration testing and added TravisCI badge (corresponding to `master` branch) to README.md

## Notes
Added `.travis.yml` for TravisCI setup by following the model presented on the Trello card and making some tweaks such as:
- including a `gem install bundler` command in the `before_install` section
- removed the original `cp` command which resulted in being unnecessary 
- removed the SCSS linting directive since these projects won't be using SCSS and is therefore not necessary
- modified the rubocop execution command due to the changes that were discussed in #2 's **Notes** 

## Screenshots
travis-ci-pr-screenshot

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+env:
+  global:
+    - RUBY_GC_MALLOC_LIMIT=90000000
+    - RUBY_GC_HEAP_FREE_SLOTS=200000
+    - SECURITY_SECRET_TOKEN=traviscitraviscitraviscitraviscitravisci
+
+services:
+  - postgresql
+  - redis-server
```
Comment (mcolombo87) [inline .travis.yml]: You don't have redis in your project, it's not necessary here

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+language: ruby
+rvm:
+  - 2.4.0
+
+addons:
+  code_climate:
```
Comment (mcolombo87) [inline .travis.yml]: Code Climate it's a great tools for analyse your code. This tool show a nice info like "Code Smell", Complexity Code (Complexity Cyclomatic), security issues, etc. But, we are not using it, so... this line is leftover. Take a look in if you want to know more of Code Climate https://codeclimate.com/

File: .travis.yml (hunk 1 above)
Comment (mcolombo87) [reply .travis.yml]: https://en.wikipedia.org/wiki/Redis and https://redis.io/

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+services:
+  - postgresql
+  - redis-server
+
+before_install:
+  - "echo '--colour' > ~/.rspec"
+  - "echo 'gem: --no-document' > ~/.gemrc"
+  # Solves rainbow issue: https://github.com/sickill/rainbow/issues/48
+  - gem update --system
+  # Solves must use Bundler 2 or greater issue: https://docs.travis-ci.com/user/languages/ruby/#bundler-20
```
Comment (mcolombo87) [inline .travis.yml]: Remove comments.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  - gem install bundler
+
+before_script:
+  - psql -c "CREATE USER \"rails-bootstrap\" WITH PASSWORD 'rails-bootstrap';" -U postgres
+  - psql -c "DROP DATABASE IF EXISTS \"rails-bootstrap_test\";" -U postgres
+  - psql -c "CREATE DATABASE \"rails-bootstrap_test\" ENCODING 'utf8';" -U postgres
+  - bundle exec rake db:migrate
+
+script:
+  - bundle exec rspec spec -fd
```
Comment (mcolombo87) [inline .travis.yml]: Rspec was not required in this card, this is a task for another moment.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  - bundle exec rubocop --require rubocop-rails app spec
+
+#notifications:
```
Comment (mcolombo87) [inline .travis.yml]: Remove comments

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+#  email:
+#    recipients:
+#      - rails-bootstrap-dev@wolox.com.ar
+
+cache:
+  bundler: true
```
Comment (mcolombo87) [inline .travis.yml]: Enter a new line here, that symbol is ugly, lol.

File: .travis.yml (hunk 1 above)
Comment (tagonzalez) [reply .travis.yml]: Removed!

File: .travis.yml (hunk 2 above)
Comment (tagonzalez) [reply .travis.yml]: Removed!

File: .travis.yml (hunk 3 above)
Comment (tagonzalez) [reply .travis.yml]: All comments removed

File: .travis.yml (hunk 4 above)
Comment (tagonzalez) [reply .travis.yml]: All references to RSpec where removed

File: .travis.yml (hunk 5 above)
Comment (tagonzalez) [reply .travis.yml]: Removed

File: .travis.yml (hunk 6 above)
Comment (tagonzalez) [reply .travis.yml]: Lol, new line entered

File: config/database.yml [hunk 7]
```diff
@@ -57,7 +57,7 @@ development:
 # Do not set this db to the same as development or production.
 test:
   <<: *default
-  database: wbooks-api_test
+  database: rails-bootstrap_test
```
Comment (mcolombo87) [inline config/database.yml]: Why this change??

Comment (tagonzalez) [reply config/database.yml]: If I understood correctly, this database name should coincide with the one dropped and created during the TravisCI configuration.
In `.travis.yml` lines 21 and 22, a database with this name is dropped (if it exists) and then created again.

It could have also been the other way around and changed the names in the `.travis.yml` file. I believe it's just a matter of the names matching for the CI test to work properly.

Comment (mcolombo87) [reply config/database.yml]: And you have right so, the properly way is changed the name in travis.yml. Your actually change is creating in every single local DB a scheme named "rails-bootstrap test" and not feel good ¿Right?

Comment (tagonzalez) [reply config/database.yml]: I agree, it would be more coherent to having it named `wbooks-api_test` since that is the name of the project/app.
Will commit the changes shortly.

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+services:
+  - postgresql
+
+before_install:
+  - "echo 'gem: --no-document' > ~/.gemrc"
+  - gem update --system
+  - gem install bundler
+
+before_script:
+  - psql -c "CREATE USER \"wbooks-api\" WITH PASSWORD 'wbooks-api';" -U postgres
```
Comment (marcegl91) [inline .travis.yml]: I'd prefer to use a separate database.travis.yml file but this is ok in this case. In your local machine, RSpec uses this test database environment in order to run your test suit, so the development database in your local machine serves as a playground where you can test your code outside of your test suit.

If you have some test database configuration that creates a conflict between RSpec (in your local computer) and TravisCI, you will be forced to add this database.travis.yml file in order to have your own custom configuration only for TravisCI.

Anyway, this is not wrong so leave it this way.

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    - RUBY_GC_HEAP_FREE_SLOTS=200000
+    - SECURITY_SECRET_TOKEN=traviscitraviscitraviscitraviscitravisci
+
+services:
+  - postgresql
+
+before_install:
+  - "echo 'gem: --no-document' > ~/.gemrc"
+  - gem update --system
+  - gem install bundler
```
[the remark this answers is not in the document]
Comment (tagonzalez) [reply .travis.yml]: The TravisCI build kept showing an error that said words to the effect of "Bundler not installed... must use Bundler 2.0 or greater...".

When researching about the error I found in the [TravisCI docs](https://docs.travis-ci.com/user/languages/ruby/#bundler-20) that this line would solve the issue and it did.

File: .travis.yml (hunk 8 above)
Comment (tagonzalez) [reply .travis.yml]: Thanks! I'll keep this in mind for when I get to setting up RSpec.
